api/libproject/devstart.rb

```diff
@@ -2282,10 +2282,10 @@ def create_or_update_workbench_db_cmd(cmd_name, args)
   :fn => ->(*args) { create_or_update_workbench_db_cmd("create-or-update-workbench-db", args) }
 })
 
-def migrate_database(gcpProject, dry_run = false)
+def migrate_database(gcc, serviceAccount = nil, dry_run = false)
   common = Common.new
   common.status "Migrating main database..."
-  CloudSqlProxyContext.new(gcpProject).run do
+  CloudSqlProxyContext.new(gcc.project, serviceAccount, gcc.creds_file).run do
```

Comment:
the project is not enough, via trial and error: need to explicitly supply the correct SA and credentials.